packages/apps/spaces/app/organization/[id]/src/components/Tabs/panels/AccountPanel/ContractNew/ContractSubtitle.tsx

```diff
@@ -37,61 +43,83 @@ export const ContractSubtitle = ({ data }: { data: Contract }) => {
     : null;
 
   const renewalPeriod = billingFrequencyOptions.find(
-    (e) => e.value === data?.contractRenewalCycle,
+    (e) => e.value === data?.billingDetails?.billingCycle,
   )?.label;
 
-  if (
-    !renewalPeriod &&
-    !hasStartedService &&
-    !serviceStartDate &&
-    data?.contractRenewalCycle
-  ) {
+  const isJustCreated =
+    DateTimeUtils.differenceInMins(
+      data.metadata.lastUpdated,
+      data.metadata.created,
+    ) === 0;
+  if (isJustCreated && !hasStartedService && !serviceStartDate) {
     return (
       <p className='font-normal shadow-none text-sm  text-gray-500 focus:text-gray-500 hover:text-gray-500 hover:no-underline focus:no-underline'>
         Contract starting...
-        <span className='underline ml-1'> Edit contract</span>
+        <Button
+          className='underline ml-1 p-0 font-normal text-sm text-gray-500 hover:text-gray-500 focus:text-gray-500'
+          variant='link'
+          size='xs'
+          onClick={onOpenEditModal}
+        >
+          Edit contract
+        </Button>
       </p>
     );
   }
-  if (!hasStartedService && !serviceStartDate && data?.contractRenewalCycle) {
+  if (
+    !hasStartedService &&
+    !serviceStartDate &&
+    data?.billingDetails?.billingCycle
+  ) {
     return (
       <p className='font-normal shadow-none text-sm  text-gray-500 focus:text-gray-500 hover:text-gray-500 hover:no-underline focus:no-underline'>
         {renewalPeriod} contract starting ...
-        <span className='underline ml-1'> Edit contract</span>
+        <Button
+          className='underline ml-1 p-0 font-normal text-sm text-gray-500 hover:text-gray-500 focus:text-gray-500'
+          variant='link'
+          size='xs'
+          onClick={onOpenEditModal}
+        >
+          Edit contract
+        </Button>
       </p>
     );
   }
-  if (!hasStartedService && serviceStartDate && data?.contractRenewalCycle) {
+  if (
+    !hasStartedService &&
+    serviceStartDate &&
+    data?.billingDetails?.billingCycle
+  ) {
     return (
-      <Text>
+      <p className='font-normal shadow-none text-sm  text-gray-500 focus:text-gray-500 hover:text-gray-500 hover:no-underline focus:no-underline'>
```

Comment:
Is there a shadow on default set on the `p` component? It shouldn't in any way. The same for `underline`.